material-ui/app/entry.server.tsx

```diff
@@ -0,0 +1,58 @@
+import * as React from 'react';
+import { renderToString } from 'react-dom/server';
+import { RemixServer } from '@remix-run/react';
+import type { EntryContext } from "@remix-run/node";
+import CssBaseline from '@mui/material/CssBaseline';
+import { ThemeProvider } from '@mui/material/styles';
+import { CacheProvider } from '@emotion/react';
+import createEmotionServer from '@emotion/server/create-instance';
+
+import createEmotionCache from './src/createEmotionCache';
+import theme from './src/theme';
+
+export default function handleRequest(
+  request: Request,
+  responseStatusCode: number,
+  responseHeaders: Headers,
+  remixContext: EntryContext,
+) {
+  const cache = createEmotionCache();
+  const { extractCriticalToChunks } = createEmotionServer(cache);
+
+  const MuiRemixServer = () => (
+    <CacheProvider value={cache}>
+      <ThemeProvider theme={theme}>
+        {/* CssBaseline kickstart an elegant, consistent, and simple baseline to build upon. */}
+        <CssBaseline />
+        <RemixServer context={remixContext} url={request.url} />
+      </ThemeProvider>
+    </CacheProvider>
+  );
+
+  // Render the component to a string.
+  const html = renderToString(<MuiRemixServer />);
+
+  // Grab the CSS from emotion
+  const { styles } = extractCriticalToChunks(html);
+
+  let stylesHTML = '';
+
+  styles.forEach(({ key, ids, css }) => {
+    const emotionKey = `${key} ${ids.join(' ')}`;
+    const newStyleTag = `<style data-emotion="${emotionKey}">${css}</style>`;
+    stylesHTML = `${stylesHTML}${newStyleTag}`;
+  });
```

Comment:
This can be simplified to

```suggestion
  const stylesHTML = styles.map(({ css, ids, key })=> {
    const emotionKey = `${key} ${ids.join(' ')}`;
    
    return `<style data-emotion="${emotionKey}">${css}</style>`;
  }).join('');
```